tests/common/test_cgroupconfigurator.py

```diff
@@ -188,6 +188,27 @@ def test_initialize_should_create_unit_files_when_the_agent_service_file_is_not_
             self.assertTrue(os.path.exists(agent_drop_in_file_cpu_accounting), "{0} was not created".format(agent_drop_in_file_cpu_accounting))
             self.assertTrue(os.path.exists(agent_drop_in_file_memory_accounting), "{0} was not created".format(agent_drop_in_file_memory_accounting))
 
+    def test_initialize_should_update_logcollector_memorylimit(self):
+        with self._get_cgroup_configurator(initialize=False) as configurator:
+            log_collector_unit_file = configurator.mocks.get_mapped_path(UnitFilePaths.logcollector)
+            original_memory_limit = "MemoryLimit=30M"
+
+            # The mock creates the slice unit file with memory limit
+            configurator.mocks.add_data_file(os.path.join(data_dir, 'init', "azure-walinuxagent-logcollector.slice"),
+                                             UnitFilePaths.logcollector)
+            self.assertTrue(os.path.exists(log_collector_unit_file),
```

Comment:
setup errors should be reported with exceptions, rather than asserts, to distinguish them from actual test verifications (such as line 208 below)

Reply:
Updated this check & the next check that memory limit exists to raise exceptions instead